src/storage/table_index_list.cpp

```diff
@@ -50,6 +53,34 @@ bool TableIndexList::NameIsUnique(const string &name) {
 	return true;
 }
 
+void TableIndexList::InitializeIndexes(ClientContext &context, DataTableInfo &table_info) {
+	lock_guard<mutex> lock(indexes_lock);
+	for (auto &index : indexes) {
+		if (index->IsUnknown()) {
+			auto &unknown_index = index->Cast<UnknownIndex>();
+			auto &index_type = unknown_index.GetIndexType();
+
+			// Do we know the type of this index now?
+			auto &catalog = Catalog::GetSystemCatalog(context);
+			auto index_type_entry = catalog.GetEntry<IndexTypeCatalogEntry>(context, DEFAULT_SCHEMA, index_type,
+			                                                                OnEntryNotFound::RETURN_NULL);
+			if (!index_type_entry) {
+				continue;
+			}
+
+			// Swap this with a new index
+			auto &create_info = unknown_index.GetCreateInfo();
+			auto &storage_info = unknown_index.GetStorageInfo();
+
+			auto index_instance = index_type_entry->create_instance(
+			    create_info.index_name, create_info.constraint_type, create_info.column_ids,
+			    unknown_index.unbound_expressions, *table_info.table_io_manager, table_info.db, storage_info);
```

Comment:
Can we run this callback function on the index instead of using the catalog entry?

Reply:
You mean pass it the unknown index? We don't always have one (e.g when creating a new index). I've been thinking of wrapping these args together into a struct thought to make it less cruddy.

My hypothesis is that we're gonna end up with more function pointers here too.